### PR TITLE
Fixes #15382 - Improved error messages on missing scenario

### PR DESCRIPTION
**Pull request by mbacovsky**
Based on users feedback I'm trying to make the error messages more clear and instructive.
The output with the patch:
```text
# foreman-installer 
ERROR: No installation scenario was selected, the installer can not continue.
       Even --help content is dependent on selected scenario.
       Select scenario with --scenario SCENARIO or list available scenarios with --list-scenarios.
```

### CODE REVIEW

**Comment by ohadlevy** [comment]
@mbacovsky note it fails on ruby 1.87

File: lib/kafo/scenario_manager.rb [hunk 1]
```diff
@@ -91,10 +91,13 @@ def select_scenario
         (available_scenarios.keys.count == 1 && available_scenarios.keys.first) ||
         select_scenario_interactively
       if scenario.nil?
-        fail_now("Scenario was not selected, can not continue. Use --list-scenarios to list available options.", :unknown_scenario)
+        fail_now("No installation scenario was selected, the installer can not continue.\n" \
```
Comment (ares) [inline lib/kafo/scenario_manager.rb]: could you please replace trailing `\` with `+` to be more explicit?

File: lib/kafo/logger.rb [hunk 2]
```diff
@@ -105,7 +105,7 @@ def dump_errors
 
     def self.dump_buffer(buffer)
       buffer.each do |log|
-        self.loggers.each { |logger| logger.send log[0], *log[1], &log[2] }
+        self.loggers.each { |logger| logger.send log[0], *([log[1]].flatten(1)), &log[2] }
```
Comment (ares) [inline lib/kafo/logger.rb]: why is this needed?

Comment (mbacovsky) [inline lib/kafo/logger.rb]: This was breaking tests on Ruby 1.8.7. In some of the log buffer records the messages were strings and not arrays and the old logger didn't recover from that.

**Comment by ares** [comment]
Thanks @mbacovsky, merging!
